subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/DefaultDecompressionCache.java

```diff
@@ -39,7 +39,7 @@ public class DefaultDecompressionCache implements DecompressionCache {
     public DefaultDecompressionCache(ScopedCacheBuilderFactory cacheBuilderFactory) {
         this.cache = cacheBuilderFactory.createCrossVersionCacheBuilder(EXPANSION_CACHE_KEY)
                 .withDisplayName(EXPANSION_CACHE_NAME)
-                .withLockOptions(mode(FileLockManager.LockMode.OnDemand))
```

Comment:
:question: Should this TODO be solved now? Or is this in the future?
I'm afraid this would be lost forever.

Reply:
It might, but it's unrelated to this PR.